packages/react-hooks/src/recordUtils.ts

```diff
@@ -14,9 +16,20 @@ import { map, filter, pipe } from 'rxjs'
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type BaseRecordAny = BaseRecord<any, any, any>
 type RecordClass<R extends BaseRecordAny> = Constructor<R> & { type: string }
+
+interface CombinedRecord<R extends BaseRecord> {
+  record: R
+  formatData: GetFormatDataReturn<[IndyCredentialFormat]>
```

Comment:
If we want to only specifically do `IndyCredentialFormat` I think we should name it accordingly. 

I also implemented this locally for some projects and just made the formatData a separate hook.

`const formattedCredential = useCredentialFormatDataById(credentialId)`

or

`const formattedCredential = useCredentialFormatData()`

Which could then optionally take a generic for the format data, i.e. `IndyCredentialFormat`. I could also create a PR with this code (and for proofs.)